src/app.js

```diff
@@ -137,20 +138,41 @@ function setupChromeEvents() {
   chrome.idle.onStateChanged.addListener(state => {
     appState.tester_state = state;
     app.pushState();
-    if (state === 'idle') {
+    if (state !== 'active') {
       checkForWorkInterval = DEFAULT_INTERVAL * 10;
+      const TEN_MINUTES = (60 * 10) * 1000;
       shutOffTimer = setTimeout(() => {
-        if (appState.tester_state === 'idle') {
+        if (appState.tester_state !== 'active') {
           appState.isPolling = false;
           app.togglePolling(appState.isPolling);
           app.pushState();
+          const ONE_MINUTE = 60 * 1000;
+          if (appState.lastRefresh > Date.now() - ONE_MINUTE) { // confirm tab recently was refreshed
+            chrome.tabs.update(appState.workTab.id, {url: ABANDONED_URL});
```

Comment:
error checking before you go ahead w/ update tab....    Rainforest extension doesn't really check to see if tabs is closed or whatever.      appState.workTab doesn't really update itself.

```javascript
chrome.tabs.get(appState.workTab.id, tab => {
  if (chrome.runtime.lastError) {
    appState.workTab = null; 
  } else {
      // your code here.. 
     
  }
});
```